### DISCUSSION TITLE
Allow snapshots of all system tables (Settings Snapshots)

### DISCUSSION THREAD

**Discussion by vancegillies** [directus/directus#15921] [Feature Requests]
It would be extremely helpful to be able to snapshot all system tables (where it makes sense) the same way we can snapshot the data model related tables.

I am currently working with Directus in a project and I am setting up permissions and roles. Now while the workflow I have come up with works it seems like this is something that the CLI should be able to be for me?

### Current Workflow
- Make permission changes while developing a feature in my local environment
- Use a SQL client to export the new row created in `directus_permissions` table
- Write a migration that creates that row
- Run bootstrap on CI to update production DB before starting the server

I haven't got into setting up flows just yet but I imagine you would have to do the same thing 

The only issue I can think of is you can't have environment specific settings as they would be wiped out when applying the snapshots, to solve this you could have different "modes" applying settings snapshots, like "replace", and "add" maybe?

This would give the option to have all the Directus settings as code and checked into your git repo (For self hosting at least). More than happy to help contribute if this is something the team/community wants!

Thanks

**Comment by azrikahar** [comment]
With https://github.com/directus/directus/discussions/15934#discussioncomment-3845556 in mind, maybe an alternative solution in the mean time would be to create to custom CLI command (such as `sync`) which makes the necessary GET requests (such as `/permissions`), and create/upsert it in the current instance so that there's no need to write custom migration with hardcoded data. 🤔

However I do also believe your particular use case is actually being discussed in #13041, so do give it an upvote 👍

**Reply by vancegillies** [reply]
I searched up and down for a discussion relating to this, seems that thread slip my sight! 

Thanks

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
